### PR TITLE
ENH add critical logger method to logging wrapper

### PR DESCRIPTION
**Pull request by beckermr**
This method is missing and I happened to need it. Adding it here.

### CODE REVIEW

**Comment by beckermr** [comment]
@rmjarvis This one is ready for review!

**Review by rmjarvis — COMMENTED** [review]
Couple things.

1. You need tests.
2. The new Mac runners on GHA enforce SIP, so you'll need to port over an equivalent of this commit: https://github.com/rmjarvis/Piff/pull/162/commits/ef788d4e79c9e69b7bc2053285d51b4cf780ab5a

**Comment by rmjarvis** [comment]
Also, out of curiosity, what kind of logging do you use critical for?  Error is already the maximum thing that I ever consider, since those are already ending the program.  Critical seems like the same thing to me for anything I do.

**Comment by beckermr** [comment]
Thanks @rmjarvis! I am using critical for some debugging. I won't be able to get to the CI changes here for a bit. Those should probably be a new PR anyways.

File: tests/test_config_image.py [hunk 1]
```diff
@@ -58,6 +58,10 @@ def test_single():
     assert logger_wrapper.isEnabledFor(logging.CRITICAL)
     assert not logger_wrapper.isEnabledFor(logging.DEBUG)
 
+    # smoke test for critical calls
+    # these are not normally used by galsim so a test here is needed
+    logger.critical("blah blah")
```
Comment (beckermr) [inline tests/test_config_image.py:63]:
```suggestion
    logger_wrapper.critical("blah blah")
```

**Comment by beckermr** [comment]
@rmjarvis I don't really understand codecov's failure here. I called the method and it is saying the actual logger call has been executed. However, it calls the if statement line partial. Any advice?

**Comment by beckermr** [comment]
ok @rmjarvis this one is ready for review!

**Review by rmjarvis — APPROVED** [review]

**Pull request merged**